app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,61 @@
+class PasswordResetsController < ApplicationController
+  before_action :get_user, only: %i(edit update)
+  before_action :valid_user, only: %i(edit update)
+  before_action :check_expiration, only: %i(edit update)
+
+  def new
+  end
+
+  def create
+    @user = User.find_by email: params[:password_reset][:email].downcase
+    if @user
+      @user.create_reset_digest
+      @user.send_password_reset_email
+      flash[:info] = t ".sent_email"
+      redirect_to root_path
+    else
+      flash.now[:danger] = t ".email_not_found"
+      render :new
+    end
+  end
+
+  def edit
+  end
+
+  def update
+    if params[:user][:password].empty?
+      @user.errors.add :password, t(".cant_be_empty")
+      render :edit
+    elsif @user.update_attributes user_params
+      log_in @user
+      flash[:success] = t ".reset_success"
+      redirect_to @user
+    else
+      render :edit
+    end
+  end
+
+  private
+
+  def user_params
+    params.require(:user).permit :password, :password_confirmation
+  end
+
+  def get_user
```

Comment:
Bad name
Follow by convention, you should not use verbs `get/set` to name method in Rails.
You can utilize other verbs such as `load/retrieve/assisgn`
For example, please change method's name to `load_user` in this case